src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -722,13 +722,16 @@ PeerImp::doProtocolStart()
 
     protocol::TMManifests tm;
 
+    auto& hashRouter = app_.getHashRouter();
+
     app_.validatorManifests ().for_each_manifest (
         [&tm](std::size_t s){tm.mutable_list()->Reserve(s);},
-        [&tm](Manifest const& manifest)
+        [&tm, &hashRouter](Manifest const& manifest)
```

Comment:
Consider tighter scoping: `[&tm, &hr = app_.getHashRouter()]` which removes the need for the stuff on line 725.

I'm 99.99% sure that C++14 makes this legal. Paging @HowardHinnant.

Reply:
FWIW, I think Nik's comment should be valid C++14 as well.